src/main/java/servlet/get/GetProfileServlet.java

```diff
@@ -23,7 +23,7 @@
 	@Override
 	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
 		Gson gson = new Gson();
-		UserBean user = new UserModel().getProfile(request.getParameter("id"));
+		UserBean user = new UserModel().getProfile(request.getParameter("username"), false);
```

Comment:
have you tried calling this servlet when you do not specify a username parameter, this looks like it would crash if the request did not contain the "username" string or if it was not properly typed. It is also questionable why we keep forcing header parameters when they are definitely not better or simpler than sending the values through body.